test/test.dart

```diff
@@ -265,3 +278,14 @@ Future<void> baasTest(
 extension RealmObjectTest on RealmObject {
   String toJson() => realmCore.objectToString(this);
```

Comment:
This is surprising `toJson` by convention should return `dynamic` (runtime-type of objects will be `Map<String, dynamic>`). I would suggest running `jsonDecode` on the decoded `String` to follow convention.

Reply:
I'm slightly worried about taking such approach because the json Core produces might not be the most valid json in the world - particularly, they'll serialize integers as strings, dates as non-roundtrippable strings, etc.